### PR TITLE
Upgrade to spigot 1.11

### PR DESCRIPTION
**Pull request by Aleksey-Terzi**

### CODE REVIEW

File: src/main/java/com/untamedears/humbug/Humbug.java [hunk 1]
```diff
@@ -2436,27 +2423,13 @@ private void hookEnderPearls() {
...
-      
-      mapStringToClass.put("ThrownEnderpearl",CustomNMSEntityEnderPearl.class);
-      mapStringToId.put("ThrownEnderpearl", Integer.valueOf(14));
-      
-      mapClassToString.put(CustomNMSEntityEnderPearl.class, "ThrownEnderpearl");
-      mapClassToId.put(CustomNMSEntityEnderPearl.class, Integer.valueOf(14));
+      RegistryMaterials<MinecraftKey, Class<? extends Entity>> b =
+    		  (RegistryMaterials<MinecraftKey, Class<? extends Entity>>)fieldB.get(null);
+
+      b.a(14, pearlKey, (Class<? extends Entity>)CustomNMSEntityEnderPearl.class);
```
Comment (ProgrammerDan) [inline src/main/java/com/untamedears/humbug/Humbug.java:2432]: Breathless, very nice.

File: src/main/java/com/untamedears/humbug/Humbug.java [hunk 2]
```diff
@@ -1666,20 +1666,6 @@ public void run() {
 	public void dropItemAtLocation(Block b, ItemStack is) {
 		dropItemAtLocation(b.getLocation(), is);
 	}
-  
-  private void damageTool(ItemStack is) {
```
Comment (ProgrammerDan) [inline src/main/java/com/untamedears/humbug/Humbug.java:1670]: I have a feeling this was used by a piece of code commented out or removed by Maxopoly's purge, some of which I need to go back and re-add, iirc.

So many things todo. 

In any case for the moment, this case remain removed. If I need it, I'll find it. This comment is mostly for me ;)

File: src/main/java/com/untamedears/humbug/Versioned.java [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
-import org.bukkit.craftbukkit.v1_10_R1.entity.CraftHorse;
-import net.minecraft.server.v1_10_R1.EntityHorse;
-import net.minecraft.server.v1_10_R1.GenericAttributes;
+import net.minecraft.server.v1_11_R1.EntityHorse;
+import net.minecraft.server.v1_11_R1.GenericAttributes;
+
+import org.bukkit.craftbukkit.v1_11_R1.entity.CraftHorse;
 import org.bukkit.entity.Entity;
 
 public class Versioned {
```
Comment (ProgrammerDan) [inline src/main/java/com/untamedears/humbug/Versioned.java:9]: I'm surprised it's still b() on the AttributeInstance, but it's good if that API has remained stable under the hood.

**Review by ProgrammerDan — APPROVED** [review]
I'm assuming you've tested the main functional components locally post-change? 

(pearl gravity looks to be the main / only one of interest to me)

**Comment by Aleksey-Terzi** [comment]
Yes, all pushed plugins (including Humbug and pearls) are tested, but I greatly appreciate if this will test someone else as well :)

File: src/main/java/com/untamedears/humbug/Versioned.java (hunk 3 above)
Comment (Aleksey-Terzi) [reply src/main/java/com/untamedears/humbug/Versioned.java:9]: This is the same method.
Actually I've compared sources of 1.10.2-spigot and 1.11.2-spigot and made sure that all "magic" methods are similar.
But I could make mistake as well and probably missed something during testing, so would be good to do some testing by someone (at least fast testing)

**Comment by ProgrammerDan** [comment]
Nice! Yeah, I've got a group of folks willing to help test, I'll be pinging them soon, thanks to you!

**Comment by ProgrammerDan** [comment]
@Aleksey-Terzi 

Although tested the build fails: 

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /var/jenkins_home/workspace/Humbug-master/src/main/java/com/untamedears/humbug/Humbug.java:[2432,75] incompatible types: java.lang.Class<com.untamedears.humbug.CustomNMSEntityEnderPearl> cannot be converted to java.lang.Class<? extends org.bukkit.entity.Entity>
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 20.704 s
[INFO] Finished at: 2017-01-05T20:03:56+00:00
[INFO] Final Memory: 26M/214M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project Humbug: Compilation failure
[ERROR] /var/jenkins_home/workspace/Humbug-master/src/main/java/com/untamedears/humbug/Humbug.java:[2432,75] incompatible types: java.lang.Class<com.untamedears.humbug.CustomNMSEntityEnderPearl> cannot be converted to java.lang.Class<? extends org.bukkit.entity.Entity>
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[JENKINS] Archiving /var/jenkins_home/workspace/Humbug-master/pom.xml to com.untamedears/Humbug/1.10.0/Humbug-1.10.0.pom
Archiving artifacts
channel stopped
Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
Finished: FAILURE
```

here: https://build.devotedmc.com/job/Humbug-master/3/console

Thoughts?

**Comment by Aleksey-Terzi** [comment]
This is bug (I confused Entity from bukkit and NMS), though for my settings I was able to compile.
I will fix this later today and push changes.

After your comment yesterday about "gravity" - I've reinvestigated code and found that gravity-related functionality was disabled at least starting from 1.10 spigot,
I'm guessing this is why Deprecation attribute was added to custom pearl's entity and item.

I will try to re-introduce it later today.

**Pull request merged**